lib/.eslintrc.yaml

```diff
@@ -15,6 +15,8 @@ rules:
     # Config specific to lib
     - selector: "NewExpression[callee.name=/Error$/]:not([callee.name=/^(AssertionError|NghttpError)$/])"
       message: "Use an error exported by the internal/errors module."
+    - selector: "CallExpression[callee.object.name='Error'][callee.property.name='captureStackTrace']"
```

Comment:
Given that there are 17 exceptions to this rule in lib, are we sure this is a good rule to have?

Reply:
It might be difficult to identify the cases which should not use `Error.captureStackTrace` without this rule.

I count 16 exceptions while 7 of those are all inside `internal/errors` and I would normally just add a file wide exception explicitly for this case but that would require to write an individual rule instead of using the `no-restricted-syntax` rule.

Three more cases would ideally also switch to using `hideStackFrames()` but C++ errors still work quite differently than our JS errors (besides N-API errors which try to mirror the JS side) and handling errors inside callbacks is not yet possible. I already have an idea to improve those further if this PR lands.

That leaves six other cases of which five seem to be exceptions to the rule in a way that they are neither directly related to our errors nor should they ever use `hideStackFrames()`.

Reply:
I updated the `hideStackFrames()` function and now there's one exception less than before.